tests/services/repository.service.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { of } from 'rxjs';
+import { RepositoryService, SESSION_AVALIABILITY_FIX_FAILED } from '../../src/app/core/services/repository.service';
+
+let repositoryService: RepositoryService;
+let githubService: any;
+
+describe('RepositoryService', () => {
+  beforeEach(() => {
+    githubService = jasmine.createSpyObj('GithubService', ['synchronizeRemoteLabels']);
+    repositoryService = new RepositoryService(null, githubService, null);
+  });
+
+  describe('.syncLabels()', () => {
+    it('should throw an error given an Observable of false', () => {
+      of(false)
+        .pipe(repositoryService.syncLabels())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(new Error(SESSION_AVALIABILITY_FIX_FAILED)),
+        });
+    });
```

Comment:
A quick question cause I'm not entirely sure,
if we supply an `of(true)` does this test fail? Or do we need to supply the `next:` in the `.subscribe()`?

So like this? Just checking cause I'm not sure haha

```typescript
it('should throw an error given an Observable of false', () => {
  of(false)
    .pipe(repositoryService.syncLabels())
    .subscribe({
      next: () => fail(),
      error: (err) => expect(err).toEqual(new Error(SESSION_AVALIABILITY_FIX_FAILED)),
  });
});
```

Reply:
Apologies Raj! 

Actually what I meant to do was the following: 
```typescript 
    it('should throw an error given an Observable of false', () => {
      of(true)
        .pipe(repositoryService.syncLabels())
        .subscribe((error: Error) => expect(error).toEqual(new Error(SESSION_AVALIABILITY_FIX_FAILED)));
    });
    ```

As a result, it would clearly describe the case where the error should be thrown when false is given

Reply:
Yea, actually I just realised with the above, it would also pass the TC if i supply a value of true :sweat_smile: 
We would need to supply the `next` to ensure that it fails! 
Thanks for noticing! ><

Reply:
No worries haha actually I realized we've made this mistake in other areas also 😆 I'll make an issue to clean that up

Reply:
Alright! :100: